torchinductor/codegen/triton.py

```diff
@@ -28,11 +29,35 @@
 from .common import IndentedBuffer
 from .common import Kernel
 from .common import OpOverrides
+from .common import SizeArg
+from .common import TensorArg
 from .common import index_prevent_reordering
 
 log = logging.getLogger(__name__)
 
 
+def signature_of(arg):
+    from triton.runtime.jit import JITFunction
+
+    if isinstance(arg, TensorArg):
+        return JITFunction._type_of(arg.dtype)
+    if isinstance(arg, SizeArg):
+        return JITFunction._key_of(V.graph.sizevars.size_hint(arg.expr))
+    raise NotImplementedError(f"unhandled {type(arg)}: {arg}")
+
+
+def config_of(args):
+    from triton.runtime.jit import JITFunction
+
+    divisible_by_16 = [
+        i
+        for i, arg in enumerate(args)
+        if isinstance(arg, TensorArg)
```

Comment:
I'm a bit concerned by saying all tensor args are divisible by 16, e.g. if there are some slicing op inside the compiled graph, and we end up calling `codegen_reference` with an odd offset https://github.com/pytorch/torchdynamo/blob/dedca39600c76a967c8881a5f1ae3aecf9fcc7be/torchinductor/ir.py#L1336

Reply:
`codegen_reference()` is only used for calling extern kernels (and return values) I believe.  For Triton kernels we apply the offset inside the indexing formula.